integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/DocumentIdBaseIT.java

```diff
@@ -314,4 +330,136 @@ public void missing() {
+		backendMock.verifyExpectationsMet();
+
+		try ( SearchSession session = mapping.createSession() ) {
+			IndexedEntity entity = new IndexedEntity( 1, "bla-bla-bla" );
+			session.indexingPlan().add( entity );
+
+			backendMock.expectWorks( INDEX_NAME )
+					.add( "fixed-prefix-1", b -> b.field( "value", "bla-bla-bla" ) )
+					.createdThenExecuted();
+		}
+		backendMock.verifyExpectationsMet();
+
+		try ( SearchSession session = mapping.createSession() ) {
+			backendMock.expectSearchReferences( Collections.singletonList( INDEX_NAME ),
+					StubSearchWorkBehavior.of( 1L, StubBackendUtils.reference( "IndexedEntity", "fixed-prefix-1" ) )
+			);
+
+			SearchQuery<EntityReference> query = session.search( IndexedEntity.class )
+					.selectEntityReference()
+					.where( f -> f.matchAll() )
+					.toQuery();
+
+			assertThat( query.fetchAll().hits() )
+					.containsExactly( EntityReferenceImpl.withName( IndexedEntity.class, "IndexedEntity", 1 ) );
+		}
+		backendMock.verifyExpectationsMet();
+	}
+
+	@Test
+	public void customBridge_withParameters_programmaticMapping() {
+		class IndexedEntity {
+			Integer id;
+			String value;
+
+			IndexedEntity(Integer id, String value) {
+				this.id = id;
+				this.value = value;
+			}
+		}
+
+		backendMock.expectSchema( INDEX_NAME, b -> b.field( "value", String.class ) );
+		SearchMapping mapping = setupHelper.start()
+				.withConfiguration( builder -> {
+					builder.addEntityType( IndexedEntity.class );
+
+					TypeMappingStep indexedEntity = builder.programmaticMapping().type( IndexedEntity.class );
+					indexedEntity.indexed().index( INDEX_NAME );
+					indexedEntity.property( "id" ).documentId().identifierBinder(
+							new ParametricBridge.ParametricBinder(),
+							Collections.singletonMap( "fixedPrefix", "fixed-prefix-" )
+					);
+					indexedEntity.property( "value" ).genericField();
+				} )
+				.expectCustomBeans().setup( IndexedEntity.class );
+		backendMock.verifyExpectationsMet();
+
+		try ( SearchSession session = mapping.createSession() ) {
+			IndexedEntity entity = new IndexedEntity( 1, "bla-bla-bla" );
+			session.indexingPlan().add( entity );
+
+			backendMock.expectWorks( INDEX_NAME )
+					.add( "fixed-prefix-1", b -> b.field( "value", "bla-bla-bla" ) )
+					.createdThenExecuted();
+		}
+		backendMock.verifyExpectationsMet();
+
+		try ( SearchSession session = mapping.createSession() ) {
+			backendMock.expectSearchReferences( Collections.singletonList( INDEX_NAME ),
+					StubSearchWorkBehavior.of( 1L, StubBackendUtils.reference( "IndexedEntity", "fixed-prefix-1" ) )
+			);
+
+			SearchQuery<EntityReference> query = session.search( IndexedEntity.class )
+					.selectEntityReference()
+					.where( f -> f.matchAll() )
+					.toQuery();
+
+			assertThat( query.fetchAll().hits() )
+					.containsExactly( EntityReferenceImpl.withName( IndexedEntity.class, "IndexedEntity", 1 ) );
+		}
+		backendMock.verifyExpectationsMet();
+	}
```

Comment:
Please also test edge cases:

* Passing a parameter that is never used by the binder (this should probably trigger an error?)
* Consuming a parameter in the binder that was not set (I suppose `.parameter()` should return null in that case... but ideally I'd align on https://hibernate.atlassian.net/browse/HSEARCH-4167. Let's discuss it?)
* Passing multiple parameters with the same name (can only happen with the annotation mapping).

Reply:
let's chat on Zulip...